### PR TITLE
Telemetry

### PR DESCRIPTION
**Pull request by tmf97**
This PR adds a telemetry module that the rest of the flight software is able to interact with (rather than calling the device drivers themselves)

### CODE REVIEW

File: drivers/power/power_controller.py [hunk 1]
```diff
@@ -503,7 +503,7 @@ def comms_amplifier(self, on: bool):
         self.set_single_output("comms", int(on), 0)
 
     def set_GPIO_low(self):
-        self.rf_switch()
+        self.rf_transmitting_switch()
```
Comment (tmf97) [inline drivers/power/power_controller.py:506]: Also want to enable receiving on rf switch

File: flight_modes/flight_mode.py [hunk 2]
```diff
@@ -93,10 +106,9 @@ def update_state(self) -> int:
 
         # go to maneuver mode if there is something in the maneuver queue
         if not self.parent.maneuver_queue.empty():
-            # TODO assign the time command to this var
-            self.burn_time = None
-            if self.burn_time - time() < (60.0 * BURN_WAIT_TIME):
-                return FMEnum.Maneuver.value
+            if params.SCHEDULED_BURN_TIME is not None and params.SCHEDULED_BURN_TIME > time():
```
Comment (tmf97) [inline flight_modes/flight_mode.py:109]: Add TODO about checking not just the maneuver queue but also the parameter

File: flight_modes/flight_mode.py [hunk 3]
```diff
@@ -383,20 +373,15 @@ def __init__(self, parent):
...
     def update_state(self):
         # check power supply to see if I can transition back to NormalMode
-        if self.parent.telemetry.gom.percent > params.EXIT_LOW_BATTERY_MODE_THRESHOLD:
-            self.parent.replace_flight_mode_by_id(FMEnum.Normal.value)
-
-        if sum(self.parent.telemetry.gom.hk.curin) > params.ENTER_ECLIPSE_MODE_CURRENT:
-            # If we do have some power coming in (i.e. we are not eclipsed by the moon/earth), reorient to face sun
-            raise NotImplementedError
+        if self.parent.telemetry.gom.percent > params.EXIT_LOW_BATTERY_MODE_THRESHOLD or \
+                sum(self.parent.telemetry.gom.hk.curin) > params.ENTER_ECLIPSE_MODE_CURRENT:
```
Comment (tmf97) [inline flight_modes/flight_mode.py:383]: Remove stuff after the "or"

File: flight_modes/flight_mode.py [hunk 4]
```diff
@@ -414,16 +399,20 @@ def __init__(self, parent):
 
     def update_state(self) -> int:
         if self.task_completed is True:
+            logger.info("Maneuver complete. Exiting maneuver mode...")
             return FMEnum.Normal.value
         return NO_FM_CHANGE
 
     def run_mode(self):
         # sleeping for 5 fewer seconds than the delay for safety
-        sleep((self.burn_time - time()) - 5)
```
Comment (tmf97) [inline flight_modes/flight_mode.py:422]: Add TODO for clearing the SCHEDULED_BURN_TIME from the parameter file

File: flight_modes/opnav_flightmode.py [hunk 5]
```diff
@@ -2,11 +2,15 @@
 from time import sleep
```
Comment (tmf97) [inline flight_modes/opnav_flightmode.py:2]: Need to clarify with @sjz38 about whether the Opnav FM defined in `flight_mode.py` or `opnav_flightmode.py` is "correct"

File: flight_modes/restart_reboot.py [hunk 6]
```diff
@@ -21,23 +20,21 @@ def __init__(self, parent):
 
     def run_mode(self):
```
Comment (tmf97) [inline flight_modes/restart_reboot.py:21]: Add TODO about differentiating between hydrogen and oxygen bootup behavior

File: main.py [hunk 7]
```diff
@@ -31,6 +26,17 @@
 from communications.downlink import DownlinkHandler
 from communications.command_definitions import CommandDefinitions
 from telemetry.telemetry import Telemetry
+from utils.boot_cause import hard_boot
+
+import queue
```
Comment (tmf97) [inline main.py:31]: remove unused import `queue`

File: requirements_pi.txt [hunk 8]
```diff
@@ -14,4 +14,5 @@ psutil
 uptime
 opencv-python-headless
 numpy-quaternion
-astropy
```
Comment (tmf97) [inline requirements_pi.txt:17]: Remove RPi.GPIO from requirements

File: telemetry/telemetry.py [hunk 9]
```diff
@@ -238,20 +244,20 @@ def standard_packet_dict(self):
...
+                'curin_2': self.gom.hk.curin[1],  # ushort
+                'curin_3': self.gom.hk.curin[2],  # ushort
+                'vboost_1': self.gom.hk.vboost[0],  # ushort
+                'vboost_2': self.gom.hk.vboost[1],  # ushort
+                'vboost_3': self.gom.hk.vboost[2],  # ushort
+                'cursys': self.gom.hk.cursys,  # ushort
+                'vbatt': self.gom.hk.vbatt,  # ushort
                 'prs_pressure': self.prs.pressure}
 
     def write_telem(self, telem):
```
Comment (tmf97) [inline telemetry/telemetry.py:263]: Add TODO

**Review by tmf97 — COMMENTED** [review]
Comments added - will change them

**Comment by tmf97** [comment]
The features implemented in this branch have been HITL tested, the procedure and test log for which can be found here: https://cornell.app.box.com/file/789148155547

File: flight_modes/opnav_flightmode.py (hunk 5 above)
Comment (sjz38) [reply flight_modes/opnav_flightmode.py:2]: The Opnav FM that we are using is the one found in flight_mode.py

**Pull request merged**